forest/jasmine/traj2stats.py

```diff
@@ -140,175 +140,712 @@ def get_nearby_locations(traj: np.ndarray) -> Tuple[dict, dict, dict]:
-    Args: traj: 2d array, output from Imp2traj(), which is a n by 8 mat, with headers as [s,x0,y0,t0,x1,y1,t1,obs]
-          where s means status (1 as flight and 0 as pause), x0,y0,t0: starting lat,lon,timestamp,
-          x1,y1,t1: ending lat,lon,timestamp,  obs (1 as observed and 0 as imputed)
-          tz_str: timezone
-          option: 'daily' or 'hourly'
-    Return: a pd dataframe, with each row as an hour/day, and each col as a feature/stat
+def gps_summaries(
+    traj: np.ndarray,
+    tz_str: str,
+    option: str,
+    places_of_interest: Union[List[str], None] = None,
+    save_log: bool = False,
+    threshold: Union[int, None] = None,
+    split_day_night: bool = False,
+    person_point_radius: float = 2,
+    place_point_radius: float = 7.5,
+) -> Tuple[pd.DataFrame, dict]:
+    """This function derives summary statistics from the imputed trajectories
+
+    If the option is hourly, it returns
+    ["year","month","day","hour","obs_duration","pause_time","flight_time","home_time",
+    "max_dist_home", "dist_traveled","av_flight_length","sd_flight_length",
+    "av_flight_duration","sd_flight_duration"]
+    if the option is daily, it additionally returns
+    ["obs_day","obs_night","radius","diameter","num_sig_places","entropy"]
+
+    Args:
+        traj: 2d array, output from Imp2traj(), which is a n by 8 mat,
+            with headers as [s,x0,y0,t0,x1,y1,t1,obs]
+            where s means status (1 as flight and 0 as pause),
+            x0,y0,t0: starting lat,lon,timestamp,
+            x1,y1,t1: ending lat,lon,timestamp,
+            obs (1 as observed and 0 as imputed)
+        tz_str: timezone
+        option: 'daily' or 'hourly'
+        places_of_interest: list of amenities or leisure places to watch,
+            keywords as used in openstreetmaps
+        save_log: bool, True if you want to output a log of locations
+            visited and their tags
+        threshold: int, time spent in a pause needs to exceed the threshold
+            to be placed in the log
+            only if save_log True, in minutes
+        split_day_night: bool, True if you want to split all metrics to
+            datetime and nighttime patterns
+            only for daily option
+        person_point_radius: float, radius of the person's circle when
+            discovering places near him in pauses
+        place_point_radius: float, radius of place's circle
+            when place is returned as centre coordinates from osm
+    Returns:
+        a pd dataframe, with each row as an hour/day,
+            and each col as a feature/stat
+        a dictionary, contains log of tags of all locations visited
+            from openstreetmap
+    Raises:
+        RuntimeError: if the query to Overpass API fails
+        ValueError: if option argument is not 'hourly' or 'daily'
     """
-    ObsTraj = traj[traj[:,7]==1,:]
-    home_x, home_y = locate_home(ObsTraj,tz_str)
-    summary_stats = []
+
+    if option not in ["hourly", "daily"]:
```

Comment:
Would it make sense to make `option` an enum? Then this check would not be necessary. Also, may be rename it to something like `frequency`?